esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -493,11 +491,19 @@ impl<F> FlashRegion<'_, F> {
     pub fn partition_size(&self) -> usize {
         self.raw.len() as _
     }
+
+    fn range(&self) -> core::ops::RangeInclusive<u32> {
+        self.raw.offset()..=self.raw.offset() + self.raw.len()
+    }
+
+    fn in_range(&self, start: u32, len: usize) -> bool {
+        self.range().contains(&start) && self.range().contains(&(start + len as u32))
```

Comment:
```suggestion
        self.range().contains(&start) && self.range().contains(&(start + len as u32 - 1))
```
It seems to me last address required to be in range is `start + len - 1`

Reply:
yes that would express the intend more clearly